tests/unit/aiplatform/test_metadata.py

```diff
@@ -0,0 +1,198 @@
+_TEST_LOCATION = "us-central1"
+_TEST_ALT_LOCATION = "europe-west4"
+_TEST_PARENT = f"projects/{_TEST_PROJECT}/locations/{_TEST_LOCATION}"
+
+# metadata_store
+_TEST_ID = "test-id"
+_TEST_DEFAULT_ID = "default"
+
+_TEST_NAME = (
+    f"projects/{_TEST_PROJECT}/locations/{_TEST_LOCATION}/metadataStores/{_TEST_ID}"
+)
+_TEST_ALT_LOC_NAME = (
+    f"projects/{_TEST_PROJECT}/locations/{_TEST_ALT_LOCATION}/metadataStores/{_TEST_ID}"
+)
+_TEST_DEFAULT_NAME = f"projects/{_TEST_PROJECT}/locations/{_TEST_LOCATION}/metadataStores/{_TEST_DEFAULT_ID}"
+
+_TEST_INVALID_NAME = f"prj/{_TEST_PROJECT}/locations/{_TEST_LOCATION}/{_TEST_ID}"
+
+# CMEK encryption
+_TEST_ENCRYPTION_KEY_NAME = "key_1234"
+_TEST_ENCRYPTION_SPEC = gca_encryption_spec.EncryptionSpec(
+    kms_key_name=_TEST_ENCRYPTION_KEY_NAME
+)
+
+
+@pytest.fixture
+def get_metadata_store_mock():
+    with patch.object(
+        MetadataServiceClient, "get_metadata_store"
+    ) as get_metadata_store_mock:
+        get_metadata_store_mock.return_value = GapicMetadataStore(
+            name=_TEST_NAME, encryption_spec=_TEST_ENCRYPTION_SPEC,
+        )
+        yield get_metadata_store_mock
+
+
+@pytest.fixture
+def get_metadata_store_without_name_mock():
+    with patch.object(
+        MetadataServiceClient, "get_metadata_store"
+    ) as get_metadata_store_mock:
+        get_metadata_store_mock.return_value = GapicMetadataStore(
+            encryption_spec=_TEST_ENCRYPTION_SPEC,
+        )
+        yield get_metadata_store_mock
+
+
+@pytest.fixture
+def create_metadata_store_mock():
+    with patch.object(
+        MetadataServiceClient, "create_metadata_store"
+    ) as create_metadata_store_mock:
+        create_metadata_store_lro_mock = mock.Mock(operation.Operation)
+        create_metadata_store_lro_mock.result.return_value = GapicMetadataStore(
+            name=_TEST_NAME, encryption_spec=_TEST_ENCRYPTION_SPEC,
+        )
+        create_metadata_store_mock.return_value = create_metadata_store_lro_mock
+        yield create_metadata_store_mock
+
+
+@pytest.fixture
+def delete_metadata_store_mock():
+    with mock.patch.object(
+        MetadataServiceClient, "delete_metadata_store"
+    ) as delete_metadata_store_mock:
+        delete_metadata_store_lro_mock = mock.Mock(operation.Operation)
+        delete_metadata_store_lro_mock.result.return_value = (
+            metadata_service.DeleteMetadataStoreRequest()
+        )
+        delete_metadata_store_mock.return_value = delete_metadata_store_lro_mock
+        yield delete_metadata_store_mock
+
+
+class TestMetadataStore:
+    def setup_method(self):
+        reload(initializer)
+        reload(aiplatform)
+
+    def teardown_method(self):
+        initializer.global_pool.shutdown(wait=True)
+
+    def test_init_metadata_store(self, get_metadata_store_mock):
+        aiplatform.init(project=_TEST_PROJECT)
+        metadata.MetadataStore(metadata_store_name=_TEST_NAME)
+        get_metadata_store_mock.assert_called_once_with(name=_TEST_NAME)
```

Comment:
For some of these test you may want to confirm `_gca_resource` is equal to the return mock `MetadataStore` like so: https://github.com/googleapis/python-aiplatform/blob/dev-test/tests/unit/aiplatform/test_datasets.py#L511

Reply:
Thanks for the pointer, done!